devtools/docker-setup.md

```diff
@@ -0,0 +1,231 @@
+# your system will return to normal.
+./devtools/setup_local.sh
+
+# Start the app. Will also take a long time
+# (but not as long as the first command)
+docker-compose up -d app frontend watch
+
+# [Optional] In the second terminal, you can tail
+# the logs of the containers
+docker-compose logs -f
+```
+
+You can navigate to three components:
+
+| Component        | Address           | Description  |
+| ------------- |:-------------:| ----- |
+| The Dashboard      | http://localhost:8002 | This is what this repository contains.<br/>By using PCF Dev, there are two users created automatically by default.<br/>Admin User: `admin`:`admin`<br/> Regular User: `user`:`pass`|
+| The mailcatcher view      | http://localhost:8025      |   Useful for debugging e-mails. There are invite flows that send e-mails. This UI captures them |
+| HTML VNC Viewer | http://localhost:6901/?password=vncpassword      | Useful for seeing Javascript Karma Tests and Selenium Tests running.<br/>Based on [this](https://github.com/ConSol/docker-headless-vnc-container) container |
+
+_Note: the credentials here are not sensitive since they are
+only used with the local deployment and the credentials are public knowledge.
+[\[1\]](https://github.com/pivotal-cf/pcfdev)
+[\[2\]](https://github.com/ConSol/docker-headless-vnc-container)_
+
+#### Working with cg-style
+
+In the event that you need to modify the styling of cg-dashboard, it will most
+likely occur in the `cg-style` repo. You can see your changes in real time by
+linking and watching for the changes there.
+
+```sh
+export CG_STYLE_PATH=/path/to/cg-style/repo/here
+docker-compose up -d watch_cg_style
+# If you're tailing the logs, you will see the `frontend` service will restart
+
+# add the link first to cg-style (ignore the initial errors)
+docker-compose exec watch bash -c "cd /cg-style && npm link && \
+  cd /cg-dashboard && npm link cloudgov-style && npm install"
+
+# restart the watch service
+docker-compose restart watch
+# after the service comes up successfully, your local changes to cg-style will
+# automatically build in the watch_cg_style service then will you will see the
+# watch service automatically rebuild.
+```
+
+##### Unlink local cg-style
+
+In the event that you want to unlink the service, just unlink and restart watch.
+
+```sh
+docker-compose exec watch bash -c "npm unlink cloudgov-style && npm install"
+docker-compose restart watch
+```
+
+#### Recompile the Backend
+
+The frontend will automatically recompile upon changing the source.
+That is not the case for the backend. Unforunately, the Go watchers
+tried caused the CPU to spike up very high.
+
+In order to recompile, just run:
+
+```sh
+docker-compose stop app
+docker-compose up -d app
+```
+
+It will replace the existing app container with a new one, thus recompiling the backend.
+
+### Full Setup: Running One-Offs
+
+Running one-offs is especially important if you don't have the proper tools
```

Comment:
I may have missed something earlier that explain this, but what are the proper tools being referenced here?